book/src/design/circuit/gadgets/ecc/var-base-scalar-mul.md

```diff
@@ -0,0 +1,258 @@
+
+The bits $\mathbf{k}_{3 \dots 1}$ are used in double-and-add using [complete addition](./addition.md#Complete-addition).
+
+If the least significant bit is set $\mathbf{k_0} = 1,$ we return the accumulator $A$. Else, if $\mathbf{k_0} = 0,$ we return $A - T$ (also using complete addition).
+
+Output $(x_{A,0}, y_{A,0})$
+
+### Circuit design
+We need six advice columns to witness $(x_T, y_T, \lambda_1, \lambda_2, x_{A,i}, \mathbf{z}_i)$. However, since $(x_T, y_T)$ are the same, we can perform two incomplete additions in a single row, reusing the same $(x_T, y_T)$. We split the scalar bits used in incomplete addition into $hi$ and $lo$ halves and process them in parallel:
+
+$$
+\begin{array}{|c|c|c|c|c|c|c|c|c|c|}
+\hline
+    x_T   &  y_T   &          z^{hi}           &    x_A^{hi}   &  \lambda_1^{hi}  &  \lambda_2^{hi}      &         z^{lo}        &  x_A^{lo}   &  \lambda_1^{lo}     &  \lambda_2^{lo}       \\\hline
+          &        &           0               &   2[T]_x      &                  &                      &   \mathbf{z}_{129}    & x_{A,129}   &                     &                       \\\hline
+    x_T   &  y_T   &    \mathbf{z}_{254}       &   x_{A,254}   & \lambda_{1,254}  & \lambda_{2,254}      &   \mathbf{z}_{128}    & x_{A,128}   & \lambda_{1,128}     & \lambda_{2,128}     \\\hline
+    x_T   &  y_T   &    \mathbf{z}_{253}       &   x_{A,253}   & \lambda_{1,253}  & \lambda_{2,253}      &   \mathbf{z}_{127}    & x_{A,127}   & \lambda_{1,127}     & \lambda_{2,127}       \\\hline
+   \vdots & \vdots &         \vdots            &    \vdots     &      \vdots      &      \vdots          &        \vdots         &  \vdots     &      \vdots         &      \vdots           \\\hline
+    x_T   &  y_T   &    \mathbf{z}_{130}       &   x_{A,130}   & \lambda_{1,130}  & \lambda_{2,130}      &   \mathbf{z}_4        & x_{A,4}     & \lambda_{1,4}       & \lambda_{2,4}         \\\hline
+    x_T   &  y_T   &    \mathbf{z}_{129}       &   x_{A,129}   & \lambda_{1,129}  & \lambda_{2,129}      &   \mathbf{z}_3        & x_{A,3}     & \lambda_{1,3}       & \lambda_{2,3}         \\\hline
+\end{array}
+$$
+
+For each $hi$ and $lo$ half, the following constraints are enforced:
+$$
+\begin{array}{|c|l|}
+\hline
+\text{Degree} & \text{Constraint} \\\hline
+2   & q_{ECC,i} \cdot (\mathbf{z}_{i+1} - 2\mathbf{z}_i - \mathbf{k}_i) = 0 \\\hline
+3 & q_{ECC,i} \cdot \mathbf{k}_i \cdot (\mathbf{k}_i - 1) = 0 \\\hline
+4 & q_{ECC,i} \cdot \left(\lambda_{1,i} \cdot (x_{A,i} - x_{P,i}) - y_{A,i} + (2\mathbf{k}_i - 1) \cdot y_{P,i}\right) = 0 \\\hline
+4 & q_{ECC,i} \cdot \left((\lambda_{1,i} + \lambda_{2,i}) \cdot (x_{A,i} - (\lambda_{1,i}^2 - x_{A,i} - x_{P,i})) - 2 y_{A,i}\right) = 0 \\\hline
+3 & q_{ECC,i} \cdot \left(\lambda_{2,i}^2 - x_{A,i+1} - (\lambda_{1,i}^2 - x_{A,i} - x_{P,i}) - x_{A,i}\right) = 0 \\\hline
+3   & q_{ECC,i} \cdot \left(\lambda_{2,i} \cdot (x_{A,i} - x_{A,i+1}) - y_{A,i} - y_{A,i+1}\right) = 0 \\\hline
```

Comment:
This constraint needs to be modified similar to the Sinsemilla gate:
- On all but the last round of incomplete addition, `y_{A,i+1}` should be substituted with its representation in terms of `lambda_{1,i+1}` etc.
- On the last round of incomplete addition, `y_{A,i+1}` should be a queried cell.